ArjunaCore/arjuna/classes/com/arjuna/ats/internal/arjuna/utils/ExecProcessId.java

```diff
@@ -141,7 +143,12 @@ public int getpid ()
                     if (tempFile != null)
                         tempFile.delete();
 
-                    StringTokenizer theTokenizer = new StringTokenizer(bstream.toString());
+                    StringTokenizer theTokenizer;
+                    try {
+                        theTokenizer = new StringTokenizer(bstream.toString(StandardCharsets.UTF_8.name()));
+                    } catch (UnsupportedEncodingException e) {
+                        throw new AssertionError("UTF-8 is not supported");
```

Comment:
AssertionErrors can be disabled from the command line so you probably need some other exception type.  The error should also be logged. Also use StandardCharsets.UTF_8.name() for consistency.